File: best-time-to-buy-and-sell-stock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit = 0
        buy = 0

        for day, price in enumerate(prices):
            profit = price - prices[buy]
            if profit > maxprofit:
                maxprofit = profit
            if prices[buy] > price:
                buy = day

        return maxprofit

File: best-time-to-buy-and-sell-stock.cpp
#include <vector>

using namespace std;

class Solution {
public:
  int maxProfit(vector<int> &prices) {
    int maxprofit = 0;
    int buy = 0;
    int len = prices.size();

    for (int i = 0; i < len; ++i) {
      int profit = prices[i] - prices[buy];
      if (profit > maxprofit) {
        maxprofit = profit;
      }
      if (prices[buy] > prices[i]) {
        buy = i;
      }
    }

    return maxprofit;
  }
};
